CRM/Event/Page/EventInfo.php

```diff
@@ -270,8 +270,9 @@ public function run() {
     );
 
     $allowRegistration = FALSE;
+    $isEventOpenForRegistration = CRM_Event_BAO_Event::validRegistrationRequest($values['event'], $this->_id);
     if (!empty($values['event']['is_online_registration'])) {
-      if (CRM_Event_BAO_Event::validRegistrationRequest($values['event'], $this->_id)) {
+      if ($isEventOpenForRegistration == 1) {
         // we always generate urls for the front end in joomla
```

Comment:
`$isEventOpenForRegistration` is boolean so I think `if ($isEventOpenForRegistration) {` is preferable.